### PR TITLE
Domain Filter Display and Dropdown

### PR DESCRIPTION
**Pull request by ceolson01**
## Description

This PR adds a domain filter dropdown, among other things.

**New stuff** ✨

* `DomainDropdown`, along with some child components
* `ColorTag` (maybe not the best name for it?)

**Changes** 🏗

* Extend the graphql `Domain` type to add local `color` and `description` fields, along with resolvers
* Make the `Select` component much more versatile

Resolves DEV-7

### CODE REVIEW

File: src/modules/core/components/ColorTag/ColorTag.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  Gold,
+  Aqua,
+  BlueGrey,
+  Purple,
+  Orange,
+  Magenta,
+  PurpleGrey,
+}
+
+const hexMap: Record<Color, string> = {
```
Comment (rdig) [inline src/modules/core/components/ColorTag/ColorTag.tsx:24]: Very nice!

File: src/modules/core/components/Fields/Select/SelectListBox.tsx [hunk 2]
```diff
@@ -42,15 +44,18 @@ const SelectListBox = ({
       )
     : -1;
   return (
+    // We're safe to disable this eslint rule - if this were a standard listbox (as opposed to collapsible, as it is now) we'd want this element to be tabbable
+    // eslint-disable-next-line jsx-a11y/aria-activedescendant-has-tabindex
```
Comment (rdig) [inline src/modules/core/components/Fields/Select/SelectListBox.tsx:48]: :+1:

File: src/modules/dashboard/components/DomainDropdown/DomainDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  }, [data]);
+
+  return (
+    <Form<FormValues>
+      initialValues={{
+        filteredDomainId: filteredDomainId ? `${filteredDomainId}` : '0',
+      }}
+      onSubmit={() => {}}
+    >
+      <Select
```
Comment (rdig) [inline src/modules/dashboard/components/DomainDropdown/DomainDropdown.tsx:134]: I'm assuming the height of the dropdown is actually determined by the `Select`.

Screenshot from 2020-11-12 17-40-40

Is there any way to make it taller?

File: src/modules/dashboard/components/ColonyHome/ColonyHome.tsx [hunk 4]
```diff
@@ -140,6 +141,9 @@ const ColonyHome = ({ match, location }: Props) => {
         </aside>
         <div className={styles.mainContent}>
           <ColonyTotalFunds colony={colony} />
+          <div className={styles.domainsDropdownContainer}>
```
Comment (rdig) [inline src/modules/dashboard/components/ColonyHome/ColonyHome.tsx:144]: Screenshot from 2020-11-12 17-39-51

When a domain is selected, we should also see that domain's description above the Colony Funds in the right sidebar.

Either add that in here, or open a new Linear issue to keep track of it _(post UAC, when we have the event data?)_

File: src/modules/dashboard/components/DomainDropdown/DomainDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            </>
+          )}
+          <div className={styles.activeItemLabel}>{label}</div>
+        </div>
+      );
+    },
+    [getDomainColor],
+  );
+
+  const options = useMemo<ComponentProps<typeof Select>['options']>(() => {
```
Comment (rdig) [inline src/modules/dashboard/components/DomainDropdown/DomainDropdown.tsx:95]: You're going nuts with typing everything. I like it! :100:

File: src/modules/dashboard/components/DomainDropdown/DomainSelectItem.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+import Heading from '~core/Heading';
+import Icon from '~core/Icon';
+import Paragraph from '~core/Paragraph';
+import { ColonyDomainsQuery } from '~data/index';
+import { ENTER } from '~types/index';
+
+import styles from './DomainSelectItem.css';
+
+interface Props {
+  domain: ColonyDomainsQuery['colony']['domains'][number];
```
Comment (rdig) [inline src/modules/dashboard/components/DomainDropdown/DomainSelectItem.tsx:19]: Maybe we should have a "proper' type for a domain? Something like `AnyDomain` ?

File: src/modules/dashboard/components/DomainDropdown/DomainSelectItem.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    ethParentDomainId,
+    name,
+  },
+}: Props) => {
+  const openUAC = useCallback(() => {
+    // fixme open UAC here
+    alert('Open UAC');
+  }, []);
+
+  const handleEditClick = useCallback<MouseEventHandler<HTMLButtonElement>>(
```
Comment (rdig) [inline src/modules/dashboard/components/DomainDropdown/DomainSelectItem.tsx:38]: Again, nice job typing these!

**Review by rdig — APPROVED** [review]
Very nice work!

I especially liked the proper way you typed most everything.

The refactor of the `Select` must have been a pain.

I've left comments here and there, but overall this is very nice work!

Screenshot from 2020-11-12 17-51-24

PS: Please make sure to rebase on the latest `feature/simplified-colony` _(let me know if you need help)_ so that #2289 which is downstream from you can also be rebased, reviewed and merged.

File: src/modules/dashboard/components/DomainDropdown/DomainSelectItem.tsx (hunk 6 above)
Comment (ceolson01) [reply src/modules/dashboard/components/DomainDropdown/DomainSelectItem.tsx]: Ah yes, nice call! There's a `OneDomain` type exported from `data` 🙂

File: src/modules/dashboard/components/ColonyHome/ColonyHome.tsx (hunk 4 above)
Comment (ceolson01) [reply src/modules/dashboard/components/ColonyHome/ColonyHome.tsx:162]: Opened!

File: src/modules/dashboard/components/DomainDropdown/DomainDropdown.tsx (hunk 3 above)
Comment (ceolson01) [reply src/modules/dashboard/components/DomainDropdown/DomainDropdown.tsx:134]: It is! Thus, I've opened an issue for this as well.

**Pull request merged**